### DISCUSSION TITLE
Ride Color Presets

### DISCUSSION THREAD

**Discussion by 733737** [OpenRCT2/OpenRCT2#18022] [Q&A] [answered]
I am creating this thread not only to discuss my PRs, but future PRs, related to Ride Color Presets. I also want to know what the community thinks about them.
List of my personal questions:
1.) Should Presets be based on Real Rides or Made Up?
2.) What Rides should receive presets?
3.) How many presets should there be?
4.) How should the pull requests be structured? ie should there be 1 ride type per PR, and how many color presets per PR?
5.) How important is this topic, and who has the most desire to implement this

Also please list and show photos of any real rides and explain corresponding color schemes that you want implemented in this thread so we can make a list of all desired color presets. We can then implement several in a PR somewhere down the line...

**Comment by Broxzier** [comment]
I don't think the game needs any more colour presets for any ride. The presets act more like placeholders, most players will recolour the rides anyway. For flat rides, the chosen preset is randomly picked, so some variance is already present if the player does not recolour their rides.

**Reply by 733737** [reply]
I agree that we should avoid cluttering the codebase and limit the number of presets, and also that too many presets may cause slowdown or network desyncs when placing a ride. However, some rides in the game could benefit from more presets, most rides have 3 or more presets, imo all rides should have at least 5 presets, but not more than 10, currently the flying saucer has 2. I want these to be one and done changes, meaning NOT adding a constant stream of random color presets and slowing down / cluttering code for the game

**Comment by ocalhoun6** [comment]
> 1.) Should Presets be based on Real Rides or Made Up?

I'd prefer presets based on real rides.

> 2.) What Rides should receive presets?

Any that have too few currently.

> 3.) How many presets should there be?

I'd say at least 5 per ride, and up to 10. No more than 10, though.

> 4.) How should the pull requests be structured? ie should there be 1 ride type per PR, and how many color presets per PR?

Definitely just one ride type per PR ... and probably best to do _all_ the presets for that ride in one PR.

> 5.) How important is this topic, and who has the most desire to implement this

Personally, I'm really excited to get new color presets in the game.

Broxzier says that most players will recolor the rides anyway, but that's not completely true for me. At least during ordinary scenario play. Unless I have some particular color scheme already going in the scenario, I typically recolor tracked rides, but let almost all flat rides (and tower rides) keep their default preset colors. I'd enjoy getting new color presets, especially for flat rides. And even for the tracked rides, I usually recolor it _after_ finishing construction, so new colors _during_ construction would make things slightly more entertaining for me personally.

I don't know enough about the codebase and coding in general to know if adding more presets is really going to slow down the code in any significant way, though. Seems like it should really be no big deal, but what do I know?

Maybe as an alternative to additional color presets, the game could assign random colors once all the presets are already used?

**Reply by 733737** [reply]
random colors have been discussed and it was decided it was a bad idea. The developers want to avoid any decrease in performance even if miniscule, especially network performance, I am glad we agree on the PR structure I outlined and utilized contemporaneously

**Comment by 733737** [comment]
When I say it may effect performance I mean two things 1.) RNG is a deterministic calculation which may cause network desyncs(please correct me if this is incorrect) 2.) low end hardware may take an extra second to place a ride if you have 10 color presets to choose from (again correct me if wrong) I like the 5-10 limit, and also don't take the time to color a ride unless the preset bothers me, or I'm doing special decorations

**Reply by ocalhoun6** [reply]
>1.) RNG is a deterministic calculation which may cause network desyncs(please correct me if this is incorrect)

You mean non-deterministic, right? As long as it's deterministic, each client should come up with the same result, so no desync.

But OpenRCT2 already includes _lots_ of random number generations (like for determining ride breakdowns, peep spawning, or certain peep decisions), and those random numbers are successfully synced in network play already. So that seems like a problem that has already been solved.

**Reply by 733737** [reply]
I didn't say that it would break multiplayer, I said it may generate more desync errors because the game has quirks

**Comment by 733737** [comment, marked as answer]
This is being actively improved
